csi_camera.cpp

```diff
@@ -81,8 +81,22 @@ Camera::raw_image CSICamera::get_image(const std::string mime_type, const Attrib
 }
 
 Camera::image_collection CSICamera::get_images() {
-    std::cerr << "get_images not implemented" << std::endl;
-    return image_collection{};
+    if (debug) {
+        std::cout << "hit get_images" << std::endl;
+    }
+    
+    AttributeMap empty_extra;
+    raw_image image = get_image(DEFAULT_OUTPUT_MIMETYPE, empty_extra);
+    image.source_name = ""; // empty string because we don't have multiple sources to differentiate
+
+    image_collection collection;
+    collection.images = std::vector<raw_image>{std::move(image)};
+    auto now = std::chrono::system_clock::now();
+    auto duration_since_epoch = now.time_since_epoch();
+    auto nanoseconds = std::chrono::duration_cast<std::chrono::nanoseconds>(duration_since_epoch);
+    collection.metadata.captured_at = std::chrono::time_point<long long, std::chrono::nanoseconds>(nanoseconds);
```

Comment:
Should this be `std::chrono::time_point<std::chrono::system_clock, std::chrono::nanoseconds>(nanoseconds)`

`<std::chrono::system_clock, std::chrono::nanoseconds>` instead of `<long long, std::chrono::nanoseconds>(nanoseconds)`

Or do we actually want long long?

Reply:
ok gotcha, ok with me to keep consistent.

just seemed a little off looking at the docs: https://en.cppreference.com/w/cpp/chrono/time_point.html

Reply:
Tomorrow I can try testing with this more idiomatic-with-docs wat